### PR TITLE
MM demo, added some images

### PR DESCRIPTION
**Pull request by mattradz**
- check image quality if you can--> from screenshots

- will add MM image on monday after demo is finished

### CODE REVIEW

File: demos/MiddleMan.py [hunk 1]
```diff
@@ -132,11 +116,7 @@ def pltImage(img_bits, img):
     eve = Eve(program)
 
     #connect agents
-    QConnect(alice, bob)
-    QConnect(bob, charlie)
-    QConnect(alice, charlie)
-    QConnect(alice, eve)
-    QConnect(bob, eve)
+    QConnect(alice, bob, charlie, eve)
```
Comment (zacespinosa) [inline demos/MiddleMan.py]: nice, this looks so much cleaner

File: docs/demos/middle-man.rst [hunk 2]
```diff
@@ -0,0 +1,259 @@
+.. _middle-man: 
+
+=========================================================
+Middle-Man Attack
+=========================================================
+
+Middle-Man Attack is a demonstration of quantum networks resistance against intruders. This demo 
+extends :ref:`superdense coding <superdense-coding>` by allowing two agents, Alice and Bob, to send 
+numerous classical bits using bell state pairs. However, in this protocol, there is an intruder agent,
+Eve, who attempts to interecept and measure the information.
```
Comment (zacespinosa) [inline docs/demos/middle-man.rst]: Maybe add a last sentence saying why quantum networks are resistant against intruders..."Despite successfully intercepting Alice and Bob's message, due to ..... , Eve will only measure random noise and Bob will be able to discern that an intruder has intercepted the message."

File: docs/demos/middle-man.rst [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+Protocol
+=========================================================
+The middle-man attack involves 4 agents, Alice, Bob, Charlie, and Eve.
+Following the :ref:`superdense coding <superdense-coding>` protocol,
+Charlie prepares the bell state pair and distributes the entangled qubits to Alice and Bob,
+where Alice operates on her bell state pair from Charlie based on the classical bits
+she wishes to send to Bob, and then sends her bell state pair to Bob. This process can be repeated
+for any number of classical bits that Alice wishes to send. 
+
+The attack occurs when Eve intercepts the Alice's qubits on the way to Bob, measures the qubits,
```
Comment (zacespinosa) [inline docs/demos/middle-man.rst]: omit `the` between intercepts and Alice's

File: docs/demos/middle-man.rst [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+    import matplotlib.image as image
+
+    img = image.imread("./Images/Logo.jpeg")
+    img_bits = list(np.unpackbits(img))
+
+    program = Program()
+    ro = program.declare('ro', 'BIT', 2*len(img_bits))
+
+Now, we can create our agents, giving charlie control over all the qubits, alice the image as
```
Comment (zacespinosa) [inline docs/demos/middle-man.rst]: Charlie*, Alice*

File: docs/demos/middle-man.rst [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+    '''
+    def run(self):
+        p = self.program
+        for i in range(0,len(alice.cmem),2):
+            p += H(i)
+            p += CNOT(i, i+1)
+            self.qsend(alice.name, [i])
+            self.qsend(bob.name, [i+1])
+
+Now, we will create Agent Alice, Eve, and Bob. Alice will operate on her bell state pair from Charlie based on the
```
Comment (zacespinosa) [inline docs/demos/middle-man.rst]: We should do either Agent or agent everywhere

File: docs/demos/middle-man.rst [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+classical bits she wishes to send to Bob. Then, she will send her qubit to Bob, but unknowingly the qubit
+is intercepted by Agent Eve. Eve is an intruder, and simply measures the intercepted information, before retransmitting
```
Comment (zacespinosa) [inline docs/demos/middle-man.rst]: refer to them always as `Agent Eve` or just `Eve`

File: docs/demos/middle-man.rst [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+does not share a bell state pair with Alice, she only measures random noise. Further, her measurements
+collapse the qubits before Bob measures them giving him only half of the information and alerting him
+to an intruder.
+
+This will be included once the demo is done:
+.. image:: ../images/Circuits/middle-man-demo.png
+
+Extend Simulation
+----------------------------------------
+You have now created a program to simulate a middle man attack! You are able to send simulate an intruder
```
Comment (zacespinosa) [inline docs/demos/middle-man.rst]: omit send

File: docs/demos/middle-man.rst [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+trying to intercept qubits of information. Can you extend this demo to include quantum error correction, 
+a smarter intruder, or even devices with realistic noise?
+
+Source Code
+=========================================================
+The source code for this demo is included in the demos directory of the netQuil repository.
```
Comment (zacespinosa) [inline docs/demos/middle-man.rst]: Add sentence telling them to checkout a difference demo or read the whitepaper?

File: docs/demos/superdense-coding.rst [hunk 9]
```diff
@@ -190,4 +187,6 @@ Extend Simulation
 You have now created a program to simulate superdense coding! You are able to send two classical bits using only one
 quantum bit. It is now time to get creative. Add noise, add extra agents, or add more classical bits.
 
-
+Source Code
+=========================================================
+The source code for this demo is included in the demos directory of the netQuil repository.
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: The source code for this demo can be found in the netQuil `repository <https://github.com/att-innovate/netQuil>`_

File: docs/demos/superdense-coding.rst [hunk 10]
```diff
@@ -151,9 +150,7 @@ in order to make the Bell State Pair, and Alice's classical memory, :math:`\text
     charlie = Charlie(program, qubits=[0,1])
 
     # Connect Agents
-    QConnect(alice, bob)
-    QConnect(bob, charlie)
-    QConnect(alice, charlie)
+    QConnect(alice, bob, charlie)
 
     # Simulate Agents
     Simulation(alice,charlie,bob).run(trials=1, agent_classes=[Alice, Charlie, Bob])
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: I think we can remove trials=1 and agent_classes

File: docs/demos/superdense-coding.rst [hunk 11]
```diff
@@ -61,10 +60,10 @@ Import Dependencies
 
 Setup Agents 
 ----------------------------------------
-Let us first define Agent Charlie who creates and distributes the Bell State Pair to Alice and Bob. We can extend the Agent
-classes and redefine our :math:`\textit{run()}` methods. To create our Bell State Pair, he can use a
+Let us first define Agent Charlie who creates and distributes the bell state pair to Alice and Bob. We can extend the Agent
+classes and redefine our :math:`\textit{run()}` methods. To create our bell State Pair, he can use a
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: To create our bell state pair*

File: docs/demos/superdense-coding.rst [hunk 12]
```diff
@@ -12,21 +12,20 @@ This protocol enables quantum computers to interact as a network in sharing clas
...
 and measures each qubit to recreate Alice's classical bits. 
 
+Circuit
+----------------------------------------
+.. image:: ../images/Circuits/superdense.png
+
 Steps 
 ----------------------------------------
-1. Charlie creates Bell State Pair using a Hadamard (:math:`\textbf{H}`) and Controlled-Not (:math:`\textbf{CNOT}`) gate,
+1. Charlie creates bell state pair using a Hadamard (:math:`\textbf{H}`) and Controlled-Not (:math:`\textbf{CNOT}`) gate,
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: Charlie create a bell state pair

File: docs/demos/middle-man.rst (hunk 5 above)
Comment (mattradz) [reply docs/demos/middle-man.rst]: changed to lowercase

**Pull request merged**